### PR TITLE
[6.0] Fix SubstConstants: serialize ErgoTree size

### LINKED ISSUE #994: Fix SubstContants: serialize ErgoTree size
This issue is to fix bug with SubstConstants operation noticed by @Luivatra.

"the ergoscript function substConstants does not include the size/contentlength uint in the output even if the input bytes have it defined"

### PR DESCRIPTION
**Pull request by aslesarenko**
Closes #994

### CODE REVIEW

**Review by kushti — CHANGES_REQUESTED** [review]
This PR contains serious changes to another 6.0 method, serialize ( introducing CheckMinimalErgoTreeVersion rule). Those changes should be moved to corresponding PR.

File: sc/shared/src/test/scala/sigmastate/lang/SigmaBinderTest.scala [hunk 1]
```diff
@@ -31,9 +31,6 @@ class SigmaBinderTest extends AnyPropSpec with ScalaCheckPropertyChecks with Mat
     res
   }
 
-  /** Checks that parsing and binding results in the expected value.
-    * @return the inferred type of the expression
-    */
```
Comment (aslesarenko) [inline sc/shared/src/test/scala/sigmastate/lang/SigmaBinderTest.scala:36]: Why the scaladoc is removed?

**Review by aslesarenko — COMMENTED** [review]
LGTM, but please address comments and consider it approved (github doesn't allow approval of own PR)

File: sc/shared/src/test/scala/sigmastate/lang/SigmaBinderTest.scala (hunk 1 above)
Comment (kushti) [reply sc/shared/src/test/scala/sigmastate/lang/SigmaBinderTest.scala]: restored

**Review by kushti — APPROVED** [review]

**Pull request merged**